app/features/transactions/transaction-hooks.ts

```diff
@@ -81,11 +233,46 @@ export function useTransactions() {
       };
     },
     getNextPageParam: (lastPage) => lastPage.nextCursor,
+    staleTime: Number.POSITIVE_INFINITY,
+    refetchOnWindowFocus: 'always',
+    refetchOnReconnect: 'always',
+    retry: 1,
   });
 
   return result;
 }
 
+export function useHasUnacknowledgedTransactions() {
+  const transactionRepository = useTransactionRepository();
+  const userId = useUser((user) => user.id);
+
+  return useQuery({
+    queryKey: [unacknowledgedTransactionsCountQueryKey],
+    queryFn: () =>
+      transactionRepository.countUnacknowledgedTransactions({ userId }),
+    select: (data) => data > 0,
+    staleTime: Number.POSITIVE_INFINITY,
+    refetchOnWindowFocus: 'always',
+    refetchOnReconnect: 'always',
+    retry: 1,
+  });
+}
+
+export function useAcknowledgeTransaction() {
+  const transactionRepository = useTransactionRepository();
+  const userId = useUser((user) => user.id);
+
+  return useMutation({
+    mutationFn: async ({ transaction }: { transaction: Transaction }) => {
```

Comment:
pls add mutation scope. it ensures that there are no two mutations of the same scope running in parallel

Reply:
I was thnking if we can have txn id to be scope id so you can have ack running in paralled for the same transaction. I forgot that you can't have dynamic scope id so forget about that.